Components: Refactor `withFocusReturn` tests to RTL

tyxla opened the pull request:
## What?
This PR refactors the `withFocusReturn` tests to fully use `@testing-library/react` instead of `react-test-renderer`.

Part of #44780.

## Why?
It is a part of the recent effort to use `@testing-library/react` as the primary testing library.

## How?
We're updating rendering to now be with RTL.

## Testing Instructions
Verify tests still pass: `npm run test:unit packages/components/src/higher-order/with-focus-return/test/index.js`

tyxla committed:
Components: Refactor withFocusReturn tests to RTL

tyxla committed:
Add a CHANGELOG entry

Mamaduka reviewed (approved):
Thanks, @tyxla! The changes here look good to me.

Suggestion (non-blocking): I was wondering if we should also refactor from `fireEvent` to `userEvent`, and help with efforts in #44952.

tyxla committed:
Remove unnecessary fireEvent

tyxla commented:
> Suggestion (non-blocking): I was wondering if we should also refactor from `fireEvent` to `userEvent`, and help with efforts in #44952.

Sure thing! Actually, the `fireEvent()` was not necessary for this specific test, so I've refactored it further to remove the `fireEvent` usage.

Mamaduka commented:
Thanks again, @tyxla!

ciampo commented:
> > Suggestion (non-blocking): I was wondering if we should also refactor from `fireEvent` to `userEvent`, and help with efforts in #44952.
> 
> Sure thing! Actually, the `fireEvent()` was not necessary for this specific test, so I've refactored it further to remove the `fireEvent` usage.

Focusing the `textarea` imperatively goes a bit against the principles that we've been adopting in our tests throughout the repo (i.e. testinfrom the point of view of the user) — in that sense, my preference would be to focus the `textarea` by clicking it / tabbing into it. WDYT ?

tyxla committed:
Click into the textarea to focus it

tyxla commented:
> > > Suggestion (non-blocking): I was wondering if we should also refactor from `fireEvent` to `userEvent`, and help with efforts in #44952.
> > 
> > 
> > Sure thing! Actually, the `fireEvent()` was not necessary for this specific test, so I've refactored it further to remove the `fireEvent` usage.
> 
> Focusing the `textarea` imperatively goes a bit against the principles that we've been adopting in our tests throughout the repo (i.e. testinfrom the point of view of the user) — in that sense, my preference would be to focus the `textarea` by clicking it / tabbing into it. WDYT ?

This actually makes a lot of sense and I agree with the sentiment. In that case, we'll need `user-event` after all. 

I've changed it in https://github.com/WordPress/gutenberg/pull/45012/commits/24b2f4e72fb2648c2c34185c512ebeb63aae5f73 to actually resemble a user click on the textarea, which consequently focuses it.

tyxla commented:
Thank you @ciampo and @Mamaduka 🙌
